src/hooks/useContainer.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+import { Point } from '../types'
+import { pointFromEvent } from '../util'
+
+const ContainerContext = React.createContext<
+  React.MutableRefObject<HTMLDivElement | null> | undefined
+>(undefined)
+ContainerContext.displayName = 'ContainerContext'
+
+export function ContainerDiv(
+  props: Omit<
```

Comment:
These types are hard for my small brain to parse. :)

Reply:
ah yes, for that I copied what VS Code said was the type of `<div>` (the `React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>` part)

the `Omit<T, 'ref'>` is to enforce that you (as a consumer of the component) don't provide the `ref` prop, which is the important bit that the component sets